### ISSUE TITLE
Generated constructors incorrectly accept positional arguments

### ISSUE THREAD

**Issue by erictraut** [dropbox/mypy-protobuf#71]
The annotated `__init__` method generated for a protobuf message accepts positional arguments, but the underlying implementation accepts only named arguments. The first parameter after `self` should be a nameless `*` parameter. This will tell mypy and other type checkers that positional arguments are not allowed when calling the constructor.

For example, a protobuf message with a `string: name` field will generate the following annotated stub:
```
class MyMessage(google___protobuf___message___Message):
    name = ... # type: typing___Text

    def __init__(self,
        name : typing___Optional[typing___Text] = None) -> None: ...
```

If callers attempt to construct this message by calling `MyMessage('bob')`, a runtime error occurs indicating that positional arguments are not allowed for this call.

To match the underlying implementation, it should be:
```
    def __init__(self, *,
        name : typing___Optional[typing___Text] = None) -> None: ...
```

**Comment by gvanrossum** [comment]
Good idea. Can you submit a patch for this? (IIUC we care more about the Python generator than about the Go one.)

**Comment by erictraut** [comment]
I was going to write a patch as suggested. Before changing any code, I ran the test script, and I'm seeing failures. Are the tests supposed to be in a passing state, or is this a known issue? Maybe it's a problem with my configuration?

Here's a snippet of the output I see when I run `run_test.sh`. I'm using MacOS and Python 3.6.
```
        with pytest.raises(ValueError, match="Unknown field garbage."):
>           s_untyped.HasField("garbage")
E           ValueError: Protocol message Simple1 has no field garbage.

test/test_generated_mypy.py:117: ValueError

...
        # Erase the types to verify that incorrect inputs fail at runtime
        # Each test here should be duplicated in test_negative to ensure mypy fails it too
        s_untyped = s  # type: Any
        with pytest.raises(ValueError, match="Unknown field garbage."):
>           s_untyped.HasField(u"garbage")
E           ValueError: Protocol message SimpleProto3 has no field garbage.

test/test_generated_mypy.py:135: ValueError
```

**Comment by gvanrossum** [comment]
@nipunn1313: Any idea what could cause this? I can't even start trying to repro these since my env is broken in other ways.

@erictraut: Did you read the comments in run_test.sh ("Uncomment these to run on mac")?

**Comment by erictraut** [comment]
I've gone far enough down the rabbit hole trying to get this to work on my system. It should be a trivial code fix for someone who has the build & test environment working.

**Comment by nipunn1313** [comment]
Hey @erictraut - I have the setup working on my machine - so could take a look here.

I am curious what issue you were having with your setup.

The "Uncomment these to run on mac" lines actually don't appear necessary any more if you have a python3 installation explicitly on mac (which exists nowadays!). I will remove those from the run_test.sh script!

**Comment by erictraut** [comment]
When I ran the test script unmodified (without uncommenting those lines), I received the errors I pasted above. This was using a Python 3.6 virtual environment.

**Comment by nipunn1313** [comment]
I put up a pull request to take care of this! Thanks!

I've been running it locally in a python2 virtualenvironment and it's been working.
It's configured to run w/ both python2 and python3 virtual environments in travis, but something must be off about that configuration as I'm able to repro the issue you described.
Will poke at it afterwards.

Will file an issue for it.

**Comment by erictraut** [comment]
Thanks! Much appreciated.

**Comment by nipunn1313** [comment]
This generated constructors issue should be fixed with #74.

The virtualenvironment issue was quite interesting and should be fixed with #76. Didn't actually have to do with py2/py3, but rather with a new release of the python library python-protobuf (this project didn't pin the version!). My setup continued to work because I still had the old version installed.
